lib/Transforms/IPO/StructFieldCacheAnalysisImpl.h

```diff
@@ -120,20 +135,40 @@ class StructFieldAccessManager {
     return LookupBFI(*Func)->getBlockProfileCount(BB);
   }
 
-  /// Retrive a pair of information if the instruction is accessing any struct
-  /// type and field number
-  Optional<StructInfoMapPairType>
-  getFieldAccessOnInstruction(const Instruction *I) const;
+  /// Retrieve branch probability information of a branch
+  Optional<double> getBranchProbability(const BasicBlock *FromBB,
+                                        const BasicBlock *ToBB) const {
+    assert(FromBB->getParent() == ToBB->getParent());
+    Function *func = const_cast<Function *>(FromBB->getParent());
+    auto Prob = LookupBPI(*func)->getEdgeProbability(FromBB, ToBB);
+    return 1.0 * Prob.getNumerator() / Prob.getDenominator();
```

Comment:
I know. But in my case, the only place I am using this function is when calculating execution count and ExecutionCountType is double now. That's why I make it return a double at the beginning. I can make this function return a BPI and do the floating point calculation out of the function to make it easier to switch to integer operations in later version.